reporting_app/templates/untabbed_datatables.html

```diff
@@ -20,4 +20,270 @@
+      });
+
+      min_date = moment.min(date_array)
+      max_date = moment.max(date_array)
+      min_week = min_date.format('W')
+      min_month = min_date.format('M')
+      min_year = min_date.format('Y')
+      var number_of_weeks = max_date.diff(min_date, 'weeks') + 1
+      var number_of_months = max_date.diff(min_date, 'months') + 1
+      var week_range = Array.apply(null, {length: number_of_weeks}).map(Number.call, Number)
+      var month_range = Array.apply(null, {length: number_of_months}).map(Number.call, Number)
+      var weeks = {}
+      var months = {}
+
+      current_week = min_week
+      current_year = min_year
+
+      var populate_blank_weeks = week_range.map(function(t){
+            if (current_week < 52) {
+                  weeks[([parseInt(current_week), parseInt(current_year)])] = 0
+                  current_week = parseInt(current_week, 10) + 1
+
+            } else if (current_week == 52) {
+                  weeks[([parseInt(current_week), parseInt(current_year)])] = 0
+                  current_week = 1
+                  current_year = parseInt(current_year, 10) + 1
+            }
+      });
+
+      current_month = min_month
+      current_year = min_year
+
+
+      var populate_blank_months = month_range.map(function(t){
+            if (current_month < 12) {
+                  months[([parseInt(current_month), parseInt(current_year)])] = 0
+                  current_month = parseInt(current_month, 10) + 1
+
+            } else if (current_month == 12) {
+                  months[([parseInt(current_month), parseInt(current_year)])] = 0
+                  current_month = 1
+                  current_year = parseInt(current_year, 10) + 1
+            }
+      });
+
+
+      var aggregate_weeks = {}
+      var aggregate_months = {}
+
+      var available_dates = data.map(function(t){
+            week_number = moment(new Date(t[0])).week()
+            month_number = (moment(new Date(t[0])).month()) + 1
+            year_number = moment(new Date(t[0])).year()
+
+            var month_year = [month_number, year_number]
+            var week_year = [week_number, year_number]
+
+            if(typeof aggregate_weeks[week_year] == 'undefined'){
+                aggregate_weeks[week_year] = t[1]
+            } else {
+                for(var i = 0; i < ((t[1]).length); i++) {
+                    aggregate_weeks[week_year][i] = aggregate_weeks[week_year][i] + t[1][i]
+                }
+            }
+            if(typeof aggregate_months[month_year] == 'undefined'){
+                aggregate_months[month_year] = t[1]
+            } else {
+                for(var i = 0; i < ((t[1]).length); i++) {
+                    aggregate_months[month_year][i] = aggregate_months[month_year][i] + t[1][i]
+                }
+            }
+      });
+
+      return [weeks, months, aggregate_weeks, aggregate_months]
+
+}
+
+
+
+function drawCharts() {
+
+      var pipeline_type = {{ pipeline | safe }};
```

Comment:
Add quotes around `{{ pipeline }}`, then you won't need to pass this in as a list of 1 item, and you won't need to pass it through `safe`.